### ISSUE TITLE
ripple effect

### ISSUE THREAD

**Issue by daimajia** [shell-software/fab#2]

**Comment by vbaidak** [comment]
Hi,

I will think about it a bit and let you know.

I don't want to use any support libraries for this implementation and in
fact currently there are no libraries which support ripple effect. That is
why implementation seems to be not very trivial and thus may take some
time.

**Comment by Narayane** [comment]
+1

Should be a great lead for you : https://github.com/traex/RippleEffect

**Comment by vbaidak** [comment]
yup, I'm investigating it to make it work starting from android ver. 9
I think it will be done soon

**Comment by wangdalong** [comment]
Hi, nice to meet you

I reviewed your Floating Action button library, 

I think it is suitable for my project,

But I am not sure how to use animation?

For example, i want scale-in , scale-out animation, but i don't know how to do.

I used xml, and also used it in code, but it didn't work.

This is my code snippet .

mFab = (FloatingActionButton)findViewById(R.id.action_button);
mFab.setAnimationOnShow(FloatingActionButton.Animations.SCALE_DOWN);

Can you help me?

**Comment by vbaidak** [comment]
Hi wangdalong,

Animations are played while calling the public methods:
- show();
- hide();
- dismiss();

There are a few questions about playing animations, so I will add this description to README.md file soon and in the next version I will also add these public methods to allow forcing the animations play:
- playAnimationOnShow();
- playAnimationOnHide();

If you need to force play the animation now and you don't want to call neither **show()**, **hide()** nor **dismiss()** try this code:

``` java
mFab.startAnimation(mFab.getAnimationOnShow()); // for animation on start
mFab.startAnimation(mFab.getAnimationOnHide()); // for animation on hide
```

**Comment by vbaidak** [comment]
Implemented in version 1.1.0
